cloud/amazon/cloudfront.py

```diff
@@ -0,0 +1,553 @@
+#!/usr/bin/python
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+DOCUMENTATION = '''
+module: cloudfront
+short_description: Create, delete and update actions for the main AWS
+  Cloudfront actions.
+description:
+  - Read the AWS documentation for Cloudfront for the correct json values
+      as there are too many to list here.
+  - Creates distributions, streaming_distributions, invalidations,
+      origin_access_ids
+  - Deletes distributions, streaming_distributions, origin_access_ids
+  - Updates distributions, streaming_distributions
+version_added: "2.1"
+requirements: [ boto3 ]
+options:
+  type:
+    description:
+      - specifies the resource to take action upon, streaming is
+        streaming_distribution
+    required: True
+    choices: [
+            'distribution',
+            'origin_access_id',
+            'invalidation',
+            'streaming',
+            ]
+    default: None
+  policy:
+    description:
+      - The path to the properly json formatted policy file or a
+        properly formatted json policy as string, see
+        https://github.com/ansible/ansible/issues/7005#issuecomment-42894813
+        on how to use it properly
+      - Used for creation and updating of resources
+    required: false
+    default: None
+  resource_id:
+    description:
+      - Required when creating an invalidation against a distribution
+      - Required for removal of distributions and origin_access_ids
+    required: false
+  state:
+    description:
+      - present to ensure resource is created or updated.
+      - absent to remove resource
+    required: false
```

Comment:
`type` is said to be the only required value. However, when I run `ansible/hacking/test-module -m ansible-modules-extras/cloud/amazon/cloudfront.py -a "type=distribution"` I get the following error.

```